src/main/java/io/opentracing/contrib/interceptors/OpenTracingInterceptor.java

```diff
@@ -89,6 +90,13 @@ public Object wrap(InvocationContext ctx) throws Exception {
             }
 
             return ctx.proceed();
+        } catch (Throwable t) {
+            logException(scope.span(), t);
+            if (t instanceof Exception) {
+                throw (Exception) t;
+            } else {
+                throw new Exception(t);
```

Comment:
Why it's wrapped? And why only for errors?

Reply:
I cannot rethrow it, as `AroundInvoke` methods can only throw `Exception`

Reply:
Then it's it better to just catch en exception? What happens if the error like OOM is caught? The type of populated exception is changed. it seems rather dangerous to do it.

Reply:
Agree, I just updated the PR to catch only Exception's